### ISSUE TITLE
Big-endian HS38 busybox runtime fail in nSIM: arc-2021.09-eng011

### ISSUE THREAD

**Issue by EvgeniiDidin** [foss-for-synopsys-dwc-arc-processors/toolchain#427]
Running vmlinux images for HS38 Big-endian built with arc-2021.09-eng011 toolchain fails with next error:

```
    HOME=/
    TERM=linux
random: fast init done

Oops
Path: /bin/busybox
CPU: 0 PID: 1 Comm: init Not tainted 5.14.3 #4
Invalid Read @ 0x00000000 by insn @ load_elf_binary+0x60c/0x1038
ECR: 0x00050100 EFA: 0x00000000 ERET: 0x80372fa8
STAT: 0x80080602 [IE K     ]   BTA: 0x8027fa04
 SP: 0x80821ea4  FP: 0x00000000 BLK: load_elf_binary+0x27a/0x1038
LPS: 0x80372fa8 LPE: 0x80372fb0 LPC: 0x040512f0
r00: 0xffffffff r01: 0x00000000 r02: 0x00000009
r03: 0x08000000 r04: 0x00020000 r05: 0x00000000
r06: 0x74006f6e r07: 0xfbdb7f97 r08: 0x0001f420
r09: 0x0001f420 r10: 0xbfb56000 r11: 0x00000be0
r12: 0x00000000

Stack Trace:
  load_elf_binary+0x60c/0x1038
Console now belongs to simulator.
```

This error reproduces both with uClibc and glibc toolchain.

Nsim command:
```
nsimdrv -prop=nsim_isa_family=av2hs -prop=nsim_isa_core=3 -prop=chipid=0xffff -prop=nsim_isa_atomic_option=1 -prop=nsim_isa_ll64_option=1 -prop=nsim_mmu=4 -prop=mmu_pagesize=8192 -prop=mmu_super_pagesize=2097152 -prop=mmu_stlb_entries=16 -prop=mmu_ntlb_ways=4 -prop=mmu_ntlb_sets=128 -prop=icache=32768,64,4,0 -prop=dcache=16384,64,2,0 -prop=nsim_isa_shift_option=2 -prop=nsim_isa_swap_option=1 -prop=nsim_isa_bitscan_option=1 -prop=nsim_isa_sat=1 -prop=nsim_isa_div_rem_option=1 -prop=nsim_isa_mpy_option=9 -prop=nsim_isa_enable_timer_0=1 -prop=nsim_isa_enable_timer_1=1 -prop=nsim_isa_number_of_interrupts=32 -prop=nsim_isa_number_of_external_interrupts=32 -prop=isa_counters=1 -prop=nsim_isa_pct_counters=8 -prop=nsim_isa_pct_size=48 -prop=nsim_isa_pct_interrupt=1 -prop=nsim_mem-dev=uart0,kind=dwuart,base=0xf0000000,irq=24 -prop=nsim_isa_aps_feature=1 -prop=nsim_isa_num_actionpoints=4 -prop=nsim_isa_rtc_option=1a  -prop=nsim_isa_big_endian=1 vmlinux
```

**Comment by abrodkin** [comment]
Supposedly the fix is https://github.com/foss-for-synopsys-dwc-arc-processors/gcc/commit/edec3e87da5d09356397d0a9c89915c070e49a36

**Comment by VVIsaev** [comment]
Can't verify fix because of #429

**Comment by VVIsaev** [comment]
Upstream gcc 11.2.0 with two patches: https://github.com/foss-for-synopsys-dwc-arc-processors/gcc/commit/edec3e87da5d09356397d0a9c89915c070e49a36 and https://github.com/foss-for-synopsys-dwc-arc-processors/gcc/commit/2ebc5bea202ae369fb1d935be6f12da3afa28a70
Upstream binutils 2.37
Upstream glibc 2.33
buildroot 2021.08 (arc64 branch):

```
# dmesg | head
Linux version 5.14.3 (isaev@isaev-debian) (arceb-buildroot-linux-gnu-gcc.br_real (Buildroot 2021.08-20-g988d77d399) 11.2.0, GNU ld (GNU Binutils) 2.37) #2 PREEMPT Wed Oct 6 13:31:01 MSK 2021
Memory @ 80000000 [1024M] 
Memory @ 100000000 [1024M] Not used
OF: fdt: Machine model: snps,zebu_hs
earlycon: uart8250 at MMIO32 0xf0000000 (options '115200n8')
printk: bootconsole [uart8250] enabled
archs-intc	: 15 priority levels (default 1)

IDENTITY	: ARCVER [0x53] ARCNUM [0x0] CHIPID [0xffff]
processor [0]	: HS38 R3.0 (ARCv2 ISA) [Big-Endian]
# /lib/libc.so.6 
GNU C Library (Buildroot) release release version 2.33.
```
